TODOs/Classes/Models/TodoList+ModelUpdates.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  TodoList+ModelUpdates.swift
+//  TODOs
+//
+//  Created by Kevin Johnson on 5/12/20.
+//  Copyright © 2020 Kevin Johnson. All rights reserved.
+//
+
+import Foundation
+
+extension TodoList {
+
+    enum MoveResult {
+        case completedMovedToShowComplete
+    }
+
+    @discardableResult
+    func move(sIndex: Int, destination: TodoList, dIndex: Int) -> MoveResult? {
+        let todo = remove(at: sIndex)
+        return reinsert(todo: todo, destination: destination, index: dIndex)
+    }
+
+    @discardableResult
+    func remove(at index: Int) -> Todo {
+        let todo: Todo
+        if showCompleted {
+            todo = todos.remove(at: index)
+            if let oIndex = incomplete.firstIndex(where: { $0 === todo }) {
+                incomplete.remove(at: oIndex)
+            }
+        } else {
+            todo = incomplete.remove(at: index)
+            if let oIndex = todos.firstIndex(where: { $0 === todo }) {
+                todos.remove(at: oIndex)
+            }
+        }
+        return todo
+    }
+
+    func reinsert(todo: Todo, destination: TodoList, index: Int) -> MoveResult? {
+        var result: MoveResult?
+        if destination.showCompleted {
+            destination.todos.insert(todo, at: index)
+            incompleteInsert(todo: todo, index: index, destination: destination)
+        } else {
+            destination.incomplete.insert(todo, at: index)
+            todosInsert(todo: todo, index: index, destination: destination)
+            if todo.completed {
+                // Moved completed todo to list not showing completed
+                // ..for now will process move then delete
+                result = .completedMovedToShowComplete
+            }
+        }
+        return result
+    }
+
+    func todosInsert(
```

Comment:
better name, not a regular insert, this is interesting into the opposite/not-visible list